common/type.c

```diff
@@ -25,3 +29,26 @@ add_type(sqlite3 *db, const char *type)
 	assert(rc == SQLITE_DONE);
 	assert(sqlite3_finalize(stmt) == SQLITE_OK);
 }
+
+int
+type_exists(sqlite3 *db, const char *type)
+{
+	int rc, count;
+	sqlite3_stmt *stmt;
+	const char *unused;
+
+	rc = sqlite3_prepare_v2(db,
+			"SELECT COUNT(*) FROM type WHERE name=?",
```

Comment:
Why "?".

Reply:
Using parameters to prevent sql injection